docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -0,0 +1,184 @@
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in the normative portions of this document are to be interpreted as described in [IETF RFC 2119](https://tools.ietf.org/html/rfc2119). These key words may appear in lowercase and still retain their meaning unless explicitly declared as non‐normative.
+
+A conforming implementation of this protocol may provide additional functionality, but must not where explicitly disallowed or would otherwise result in non‐conformance.
+
+### Overview
+
+An _update_ is defined as a [_manifest_](#manifest-response) together with the assets referenced inside the manifest.
+Expo Updates is a protocol for assembling and delivering updates to apps running on multiple platforms.
+
+An app running a conformant Expo Updates client-library will fetch the most recent update from a conformant update server. If the client-library cannot fetch an update or if it already has the most recent update, the client-library will load the update saved in the client-library's cache.
+
+The following describes how a client-library will interact with a conforming updates server:
+1. The client-library will make a [request](#request) for the most recent manifest. 
+2. If it is a new manifest, the client-library will proceed to make requests to download and store the assets specified in the manifest.
+3. Finally, the client-library will update its local state according to any metadata provided by the response [headers](#headers).
+
+The primary consumers of this spec are Expo Application Services and organizations that wish to manage their own update server to satisfy internal requirements.
+
+## Manifest Request
+A conformant client-library MUST make a GET request with the following headers:
+```
+accept: application/json
+expo-platform: string
+expo-runtime-version: string
+```
+along with any headers stipulated by a previous responses' [server defined headers](#headers):
+
+* `expo-platform` MUST be platform type the client is running on: 
+    * iOS MUST be `expo-platform: ios`
+    * Android MUST be `expo-platform: android`
+* `expo-runtime-version` MUST be the runtime version the client is running on.
+
+A conformant client-library SHOULD also accept `application/expo+json` so that servers can easily tell when to send back an Expo manifest:
+```
+accept: application/expo+json, application/json
+```
+
+## Manifest Response
+
+A conformant server MUST return a response containing a [manifest body](#body) and [headers](#headers).
+
+The choice of manifest and headers are dependent on the values of the request headers. A conformant server MUST:
+
+* Return a manifest that describes the most recent update ordered by creation time satisfying all constraints imposed by the [request headers](#manifest-request).
```

Comment:
One thing I realized is that a server can correctly respond with a manifest that satisfies the constraints but isn't the most recent. This happens during rollouts. If a client is assigned the older of the two updates in a rollout, that is valid. (I would argue that the rollout token is not a constraint, similar to how other fields that could be used for rollouts like IP address are not constraints.)

Reply:
```suggestion
* Respond with a manifest for the most recent update, ordered by creation time, satisfying all parameters and constraints imposed by the [request headers](#manifest-request). The server MAY use any properties of the request like its headers and source IP address to choose amongst several updates that all satisfy the request's constraints.
```